src/openforms/submissions/tests/form_logic/test_binding.py

```diff
@@ -0,0 +1,375 @@
+
+DEFAULT_REQUEST_HEADERS = {
+    "Accept",
+    "Accept-Encoding",
+    "Connection",
+    "Content-Type",
+    "User-Agent",
+}
+
+
+def data_mapping_values() -> st.SearchStrategy[Any]:
+    "Generates values for the value side of openforms.typing.DataMapping"
+    return st.one_of(st.text(), st.integers(), st.floats(), st.dates(), st.datetimes())
+
+
+# the first test still takes too much time
+# decorating given because I don't want to count on test running order
+given = settings(deadline=500)(given)
+
+
+class ServiceFetchConfigVariableBindingTests(SimpleTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.service = ServiceFactory.build(
+            api_type=APITypes.orc,
+            api_root="https://httpbin.org/",
+            auth_type=AuthTypes.no_auth,
+            oas_file=FileField(
+                from_path=str(Path(__file__).parent.parent / "files" / "openapi.yaml")
+            ),
+        )
+        # prevent parsing the yaml over and over and over
+        cls.service.id = 1  # need pk for __hash__
+        cls.service.build_client = lru_cache(1)(cls.service.build_client)
+
+    @classmethod
+    def tearDownClass(cls):
+        # we did not touch the database, we're Simple
+        # super() insists on calling _remove_databases_failures
+        pass
+
+    @requests_mock.Mocker()
+    def test_it_performs_simple_get(self, m):
+        m.get("https://httpbin.org/get", json={"url": "https://httpbin.org/get"})
+        var = FormVariableFactory.build(
+            service_fetch_configuration=ServiceFetchConfigurationFactory.build(
+                service=self.service,
+                path="get",
+            )
+        )
+
+        value = bind(var, {})
+
+        self.assertEqual(value["url"], "https://httpbin.org/get")
+
+    @given(data_mapping_values())
+    @example("../otherendpoint")
+    def test_it_can_construct_simple_path_parameters(self, field_value):
+        # https://swagger.io/docs/specification/describing-parameters/#path-parameters
+        context = {"seconds": field_value}
+
+        var = FormVariableFactory.build(
+            service_fetch_configuration=ServiceFetchConfigurationFactory.build(
+                service=self.service,
+                path="delay/{seconds}",  # this is not defined as number in the OAS
+            )
+        )
+
+        with requests_mock.Mocker() as m:
+            m.get(requests_mock.ANY)
+            _ = bind(var, context)
+            request = m.last_request
+
+        head, rest = request.url[:26], request.url[26:]
+        self.assertEqual(head, "https://httpbin.org/delay/")
+        # the service should get the value
+        self.assertEqual(unquote(rest), str(field_value))
+
+        # I guess if the rest is unmangled, the rest of the assertions are not
+        # really needed; I think
+        # ∄ a way for Bobby Tables to drop a table ∧ become an unmangled entry in it
+        # but asserting is cheaper than a CVE
```

Comment:
what?

Reply:
tl;dr;
I suspect that iff `rest` is well-formed all following assertions are tautologies. But better to be safe than sorry; if being safe is this cheap.


∄ : there does not exist
Bobby Tables: https://xkcd.com/327/
∧: and

Reply:
I can't even wrap my head around what `head` and `rest` are supposed to be and why all this is happening?

And please stick to plain English in these comments, my mathematical notation knowledge is buried deep in the back of my mind, but I should be able to understand this. That can't be said from colleagues that don't have any compsci or mathematical background at all.